plugins/foreman_remote_execution/0.0/index.md

```diff
@@ -0,0 +1,375 @@
+---
+layout: plugin
+title: Foreman Remote Execution 0.0 Manual
+version: 0.0
+---
+
+# 1. {{ page.title }}
+
+This plugin enables Foreman to run arbitrary commands on hosts using
+different providers. Initially only SSH provider is supported but we
+plan to add more. Communication goes through Smart proxy so Foreman 
+does not have to have direct access to the target host and can scale
+to control many hosts. A command can be customized similarly to 
+provisioning templates or partition tables, for more details see
+[Job Templates](plugins/foreman_remote_execution/{{page.version}}/index.html#3.1JobTemplates)
+section.
+
+The command can be executed on multiple servers at once while specific
+parts of it can vary per host - in other words you can use variables
+in your commands. Variable values can be filled by given host fact,
+smart class parameter, smart variable or even host parameter. Last but
+not least, the value can be specified by user when he or she triggers
+the execution.
+
+Few examples of tasks that this plugins helps you to achieve
+
+- install/update/remove package
+- bootstrap configuration management agent
+- trigger puppet/salt/chef/... run
+
+
+## 1.1 Components
+
+The whole remote execution system is pluggable, making it easy to add
+more providers of communication later. Therefore it's split into
+several components that must be installed on Foreman  and Foreman proxy.
+
+<table class="table table-bordered table-condensed">
+  <tr>
+    <th>Plugin</th>
+    <th>Description</th>
+  </tr>
+  <tr>
+    <td>foreman_remote_execution</td>
+    <td>Required, contains the UI part and currently also SSH provider, depends on foreman-tasks</td>
+  </tr>
+  <tr>
+    <td>foreman_tasks</td>
+    <td>Required, keep status of all executed commands, allows debugging if something goes wrong</td>
+  </tr>
+  <tr>
+    <td>smart_proxy_remote_execution_ssh</td>
+    <td>Smart proxy plugin, required for using SSH provider, proxies the communication between Foreman and hosts, requires smart_proxy_dynflow</td>
+  </tr>
+  <tr>
+    <td>smart_proxy_dynflow</td>
+    <td>Smart proxy plugin, runs the jobs asynchronously and keeps status of commands executed via this proxy. This plugins uses relational SQL DB (sqlite).</td>
+  </tr>
+</table>
+
+Each component is packaged for platforms that Foreman is officially
+packaged for. Since both Foreman a Smart proxy plugins requires
+Dynflow, Ruby 1.9+ is required. Foreman 1.9+ should be compatible with
+0.0.* versions.
+
+# 2. Installation
+
+This chapter covers clean Foreman and plugins installation from
+nightly repositories on CentOS 7.1. We assume smart proxy being
+installed on the same host, it's not requirement though.
+Please adapt according to your needs.
+
+Start by installing Foreman nightly repository and EPEL7 by
```

Comment:
I think this beats the simplicity but as you wish. We'll have to write a blog post with simple example then.